recipe_box/src/main/java/com/spr/recipe_box/Service/Services/RecipeService.java

```diff
@@ -0,0 +1,145 @@
+package com.spr.recipe_box.Service.Services;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.spr.recipe_box.Constant.Constants;
+import com.spr.recipe_box.Constant.Keys;
+import com.spr.recipe_box.Model.Recipe;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+@Service
+public class RecipeService {
+    public List<Recipe> findById(String recipe_id){
+        List<Recipe> recipeList = new ArrayList<>();
+
+        String url = Constants.BIG_OVEN + Constants.RECIPE + recipe_id + Constants.API_KEY_QUERY + Keys.BIG_OVEN_KEY;

```

Comment:
I think you could refactor from line 23 to line 36 as its own method that returns a response, if you wanted to